stdlib/args.ysh

```diff
@@ -74,11 +85,18 @@ proc flag (short, long ; type='bool', default=null, help=null) {
   ##     flag -f --file ('str', help="File to process")
   ##   }
 
-  # TODO: type objects (ie. Bool, Int, Str)
-  ctx emit flags ({short, long, type, default, help})
+  # bool has a default of false, not null
+  if (type === 'bool' and default === null) {
+    setvar default = false
+  }
+
+  # TODO: Should use "trimPrefix"
+  var name = long[2:]
+
+  ctx emit flags ({short, long, name, type, default, help, required})
 }
 
-proc arg (name ; ; default=null, help=null) {
+proc arg (name ; ; default=null, help=null, required=true) {
```

Comment:
Hm what happens if you do this?

```
arg pos1 (required=false)
arg pos2 (required=true)
```

Then you have something not required before something required

I think it could make sense for all args to be required

And then the user can use `rest args` if they want their own validation



```
```